ggshield/scan/repo.py

```diff
@@ -152,33 +154,39 @@ def scan_commit_range(
     :param verbose: Display successful scan's message
     """
 
-    commits_batch = get_commits_by_batch(
-        commit_list=(
-            Commit(sha=sha, exclusion_regexes=exclusion_regexes) for sha in commit_list
+    with create_progress_bar(type_doc="commits", transient=True) as progress:
```

Comment:
Why do we need `transient` here and not on the others?

Reply:
Oh, my bad ! This came from the fact that the output handler was used in the context. The progress bar would be immediately "re-rendered" after the output was generated, resulting in two progress bars in the end.  
But by moving the output handler outside the context, everything is fine without the transient option.